#! /bin/sh /usr/share/dpatch/dpatch-run
## 06_build_dir_use_physical_path.dpatch by Andrew Ross <andrewross@users.sourceforge.net>
##
## DP: Use physical path (resolve symlinks) when checking whether in build
## DP: directory. (This patch was taken from the upstream SVN repository.)

@DPATCH@

--- plplot-5.8.0.orig/src/plcore.c
+++ plplot-5.8.0/src/plcore.c
@@ -2110,6 +2110,7 @@
 
   if (inited == 0) {
     char currdir[256];
+    char builddir[256];
 
 /* AM: getcwd has a somewhat strange status on Windows, its proper
    name is _getcwd, this is a problem in the case of DLLs, like with
@@ -2121,8 +2122,22 @@
 
     if (getcwd(currdir, 256) == NULL) {
       pldebug("plInBuildTree():", "Not enough buffer space");
-    } else if (strncmp(BUILD_DIR, currdir, strlen(BUILD_DIR)) == 0)
-      inBuildTree = 1;
+    } else {
+      /* The chdir / getcwd call is to ensure we get the physical
+       * path without any symlinks */
+      /* Ignore error in chdir - build tree may not exist */
+      if (chdir(BUILD_DIR) == 0) {
+         if (getcwd(builddir, 256) == NULL) {
+           pldebug("plInBuildTree():", "Not enough buffer space");
+         }
+         else {
+           if (strncmp(builddir, currdir, strlen(builddir)) == 0) {
+             inBuildTree = 1;
+           }
+         }
+         chdir(currdir);
+      }
+    }
     inited = 1;
   }
   return inBuildTree;
